ql/src/java/org/apache/hadoop/hive/ql/exec/ColumnInfo.java

```diff
@@ -166,9 +170,13 @@ public String getTabAlias() {
   }
 
   public boolean getIsVirtualCol() {
-    return isVirtualCol;
+    return isVirtualCol || isPartitionCol;
```

Comment:
Please don't mix virtual columns and partition columns. These are very different things. `getIsVirtualCol()` should depend on `isVirtualCol` only.

Reply:
ATM all partitionCol were virtual ones, I've added isPartitionCol to include then into Scan

Reply:
I still has the impression that this change is not safe. Example:
here we create a new instance of `ColumnInfo` based on an existing one. If the source `ColumnInfo` is a partition column the field `isPartitionCol` is still `false` in the result which is not valid.

https://github.com/apache/hive/blob/b2a5933c0f396ade25273ac75cbf35e205969b21/ql/src/java/org/apache/hadoop/hive/ql/optimizer/ColumnPrunerProcFactory.java#L196-L197

Reply:
true, but how else should I call this method? `isRequiredCol` ??

Reply:
renamed to `isHiddenPartitionCol `